tests/integration_tests/integration_settings.py

```diff
@@ -87,6 +88,13 @@
 # for the keypair.  (Defaults to pycloudlib's default behaviour.)
 KEYPAIR_NAME = None
 
+##################################################################
+# OPENSTACK SETTINGS
+##################################################################
+# Network to use for Openstack. Should be one of the names/ids found
+# in `openstack network list`
+OPENSTACK_NETWORK = None
```

Comment:
If left at its default value, this produces:

```pytb
Traceback (most recent call last):
  File "/home/daniel/dev/cloud-init/tests/integration_tests/conftest.py", line 224, in client
    with _client(request, fixture_utils, session_cloud) as client:
  File "/usr/lib/python3.8/contextlib.py", line 113, in __enter__
    return next(self.gen)
  File "/home/daniel/dev/cloud-init/tests/integration_tests/conftest.py", line 212, in _client
    with session_cloud.launch(
  File "/home/daniel/dev/cloud-init/tests/integration_tests/clouds.py", line 170, in launch
    pycloudlib_instance = self._perform_launch(kwargs)
  File "/home/daniel/dev/cloud-init/tests/integration_tests/clouds.py", line 144, in _perform_launch
    pycloudlib_instance = self.cloud_instance.launch(**launch_kwargs)
  File "/home/daniel/dev/pycloudlib/pycloudlib/openstack/cloud.py", line 98, in launch
    net = self.conn.network.find_network(self.network)
  File "/home/daniel/.virtualenvs/cloud-init/lib/python3.8/site-packages/openstack/network/v2/_proxy.py", line 1376, in find_network
    return self._find(_network.Network, name_or_id,
  File "/home/daniel/.virtualenvs/cloud-init/lib/python3.8/site-packages/openstack/proxy.py", line 369, in _find
    return resource_type.find(self, name_or_id,
  File "/home/daniel/.virtualenvs/cloud-init/lib/python3.8/site-packages/openstack/resource.py", line 1901, in find
    return match.fetch(session, **params)
  File "/home/daniel/.virtualenvs/cloud-init/lib/python3.8/site-packages/openstack/resource.py", line 1450, in fetch
    request = self._prepare_request(requires_id=requires_id,
  File "/home/daniel/.virtualenvs/cloud-init/lib/python3.8/site-packages/openstack/network/v2/_base.py", line 22, in _prepare_request
    req = super(NetworkResource, self)._prepare_request(
  File "/home/daniel/.virtualenvs/cloud-init/lib/python3.8/site-packages/openstack/resource.py", line 1136, in _prepare_request
    raise exceptions.InvalidRequest(
openstack.exceptions.InvalidRequest: Request requires an ID but none was found
```

That this error message is indicating that one should set `OPENSTACK_NETWORK` to something is not immediately obvious: should we introduce a guard somewhere that prompts people to configure this correctly before we reach this traceback?

Reply:
This too, from a quick test locally.